packages/react-query/src/scripts/utils/loadModule.ts

```diff
@@ -1,7 +1,7 @@
 export function loadModule<T>(name: string): T | undefined {
```

Comment:
How about use typescript discriminated union

```suggestion
export function loadModule<T>(name: string): { exports: T, isSuccess: true } | { exports: undefined, isSuccess: false } {
```

### AS IS

```ts
const modules = indexFileContent.matchAll(/export \* from ['"](.+?)['"]/g)

for (const [, moduleName] of modules) {
  const moduleExports = loadModule<Record<string, unknown>>(moduleName)

  if (moduleExports === undefined) {
    console.warn('[@suspensive/react-query]', `Module ${moduleName} is not found`)
    exit(1)
  }

  const { ...interfaces } = moduleExports
}
```

### TO BE

```ts
const modules = indexFileContent.matchAll(/export \* from ['"](.+?)['"]/g)

for (const [, moduleName] of modules) {
  const module = loadModule<Record<string, unknown>>(moduleName)

  if (module.isSuccess) { // in my opinion use can success easily
    const { ...interfaces } = module.exports
  }

  console.warn('[@suspensive/react-query]', `Module ${moduleName} is not found`)
  exit(1)
}
```

Reply:
Great! This will improve handling to make it easier and clearer.